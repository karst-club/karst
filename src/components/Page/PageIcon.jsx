import React from 'react';

function PageIcon(props) {
  return (
    <div className="Page-header-icon">
      <span className="Page-header-icon">{props.emoji}</span>
    </div>
  );
}

export default PageIcon;
