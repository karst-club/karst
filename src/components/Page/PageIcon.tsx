import React from 'react';
import styled from 'styled-components';

export interface Props {
  emoji: string;
}

const PageHeaderIcon = styled.div`
  display: flex;
  height: 78px;
  width: 78px;

  align-items: center;
  justify-content: center;

  align-self: flex-start;
  border-radius: 3px;
  flex-shrink: 0;
  margin-top: -24px;
  margin-left: 3px;
  margin-bottom: 0px;

  position: relative;
  z-index: 1;

  font-size: 78px;
  color: black;
  line-height: 1em;
`;

const PageIcon = ({ emoji }: Props) => (
  <PageHeaderIcon>
    <span>{emoji}</span>
  </PageHeaderIcon>
);

export default PageIcon;
