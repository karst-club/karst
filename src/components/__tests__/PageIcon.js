import React from 'react';
import renderer from 'react-test-renderer';
import PageIcon from '../PageIcon';

describe('PageIcon', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PageIcon emoji="❤️" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
