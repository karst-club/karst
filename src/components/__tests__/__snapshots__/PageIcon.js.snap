// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageIcon renders correctly 1`] = `
<div
  className="sc-bdfBwQ mEpND"
>
  <span>
    ❤️
  </span>
</div>
`;
